### PR TITLE
Add deprecation warning to fsdp_config

### PR DESCRIPTION
**Pull request by KuuCi**
Deprecates fsdp_config and replaces it with parallelism_config. Replaces all instances of the test, as well as new unit tests to make sure that the proper warnings are thrown when users attempt to use fsdp_config

### CODE REVIEW

File: llmfoundry/command_utils/eval.py [hunk 1]
```diff
@@ -65,6 +68,17 @@ def evaluate_model(
     should_log_config: bool = True,
     load_path: Optional[str] = None,
 ):
+    if fsdp_config:
+        warnings.warn(
+            VersionedDeprecationWarning(
+                'The argument fsdp_config is deprecated. Please use parallelism_config instead.',
+                remove_version='0.12.0',
```
Comment (snarayan21) [inline llmfoundry/command_utils/eval.py:75]: what's our policy on this? should this be a later version?

Comment (dakinggg) [reply llmfoundry/command_utils/eval.py:75]: yes, +2 versions

File: llmfoundry/command_utils/eval.py [hunk 2]
```diff
@@ -62,9 +64,21 @@ def evaluate_model(
...
     load_path: Optional[str] = None,
 ):
+    if fsdp_config:
+        warnings.warn(
+            VersionedDeprecationWarning(
+                'The argument fsdp_config is deprecated. Please use parallelism_config instead.',
+                remove_version='0.13.0',
+            ),
+        )
+    if fsdp_config and parallelism_config:
```
Comment (dakinggg) [inline llmfoundry/command_utils/eval.py:78]: for anyone using positional arguments, would be good to check parallelism_config and raise an error if it looks like fsdp config

Comment (KuuCi) [reply llmfoundry/command_utils/eval.py:85]: I think I did it right.....

>>> from composer.utils import dist, get_device, reproducibility, parallelism
>>> print(list(parallelism.FSDPConfig.__annotations__.keys()))
['activation_checkpointing', 'activation_checkpointing_reentrant', 'activation_cpu_offload', 'auto_wrap', 'te_checkpoint_wrapper', 'te_shard_fp8_weight', 'backward_prefetch', 'backward_prefetch_limit', 'cpu_offload', 'data_parallel_shard_degree', 'data_parallel_replicate_degree', 'device_mesh', 'forward_prefetch', 'forward_prefetch_limit', 'ignored_modules', 'keep_low_precision_grads', 'limit_all_gathers', 'load_monolith_rank0_only', 'load_planner', 'mixed_precision', 'process_group', 'save_planner', 'sharded_ckpt_prefix_dir', 'sharding_strategy', 'state_dict_type', 'sync_module_states', 'use_orig_params', 'verbose']

**Review by dakinggg — COMMENTED** [review]
can you add unit tests for calling this function with the different args and make sure the expected outcomes happen?

**Review by mvpatel2000 — COMMENTED** [review]
PR descriptions are your friend

**Review by mvpatel2000 — APPROVED** [review]

**Pull request merged**